enterprisemediawiki/meza
# Add memcached and Flow

This PR modifies install.sh to add a new script memcached.sh, which installs the memcached package, copies over a configuration file, and starts the service. This was discussed as part of #16. This PR also adds Extension:Flow. This closes #37.

#### Install Instructions:
Run install.sh on both CentOS 6 and 7 using the `flow` branch:
* `curl -LO https://raw.githubusercontent.com/enterprisemediawiki/meza/flow/scripts/install.sh`
* `sudo bash install.sh`

#### Test Instructions:
Verify memcached was installed and is running by executing `memcached-tool 127.0.0.1:11211 stats` in the command line. You should see something like the following:

```
#127.0.0.1:11211   Field       Value
         accepting_conns           1
               auth_cmds           0
             auth_errors           0
                   bytes     3180160
              bytes_read    15841082
           bytes_written    25422944
              cas_badval           0
                cas_hits           0
              cas_misses           0
               cmd_flush           0
                 cmd_get       61972
                 cmd_set       56145
               cmd_touch           0
             conn_yields           0
   connection_structures          17
        curr_connections          10
              curr_items        1797
               decr_hits           0
             decr_misses           0
             delete_hits          34
           delete_misses          14
       evicted_unfetched           0
               evictions           0
       expired_unfetched          11
                get_hits       59873
              get_misses        2099
              hash_bytes      524288
       hash_is_expanding           0
        hash_power_level          16
               incr_hits           0
             incr_misses           0
                libevent 2.0.21-stable
          limit_maxbytes   536870912
     listen_disabled_num           0
                     pid        1227
            pointer_size          64
               reclaimed          17
            reserved_fds          20
           rusage_system    3.410289
             rusage_user    1.508900
                 threads           4
                    time  1445395858
       total_connections        1286
             total_items       56137
              touch_hits           0
            touch_misses           0
                  uptime       19294
                 version      1.4.15
```

To test Flow, go to a discussion page with no content. You should see the Flow board like this:

![image](https://cloud.githubusercontent.com/assets/6728676/10645197/1e0c1470-77f2-11e5-9a0b-7e3a5f8ac2b1.png)

#### Test Results Checklist:
- [x] memcached works on CentOS 7
- [x] Flow works on CentOS 7
- [x] Update page property value and verify update with "browse properties"
- [x] Update page property value and verify update displays in `#ask` and `#show` query results
- [x] Update subobject property value and verify update with "browse properties"
- [x] Update subobject property value and verify update displays in `#ask` and `#show` query results

#### CentOS 6
This PR is not addressing CentOS 6 due to time constraints (the developers need to release for CentOS 7 ASAP). Prior to meza v1.0 these two issues must be addressed:

* memcached works on CentOS 6
* Flow works on CentOS 6

These are being tracked in #254

#### Considerations for future improvement
Note: these are not handled in this PR. See issue #253.

* See [This documentation](https://www.mediawiki.org/wiki/Flow/Converting_talk_pages) regarding conversion of existing discussion pages.
* [Extension:CLDR](https://www.mediawiki.org/wiki/Extension:CLDR) supposedly adds more friendly time references, but I haven't messed with it
* [Extension:EventLogging](https://www.mediawiki.org/wiki/Extension:EventLogging) can be used for analytics
* The settings in ExtensionSettings can be modified to enable Flow where we want. I just picked these options for now to start testing use of Flow with our wikis.
* These same settings reference `CONTENT_MODEL_FLOW_BOARD`, which throws a warning during updateExtensions.php. I'm not sure how best to address this, but it seems to install and function fine.

#### Reference
* [Tutorial](http://www.if-not-true-then-false.com/2010/install-memcached-on-centos-fedora-red-hat/)
